src/Nethermind/Nethermind.Network.Discovery/Lifecycle/NodeLifecycleManager.cs

```diff
@@ -166,25 +168,38 @@ public void ProcessNeighborsMsg(NeighborsMsg? msg)
             return;
         }
 
-        if (_isNeighborsExpected)
+        if (_lastNeighbourSize + msg.Nodes.Length == 16)
```

Comment:
What if its greater than 16? Is that possible

Reply:
In theory. 16 is enough to fix like 99% of unexpected message though.

Reply:
I mean should it be:
```suggestion
        if (_lastNeighbourSize + msg.Nodes.Length >= 16)
```